docs/designs/data_transform.md

```diff
@@ -199,14 +199,30 @@ SELECT *
 FROM census_income
 TO TRAIN WideAndDeepClassifier
 COLUMN
-    EMBEDDING(CONCAT(VOCABULARIZE(workclass), BUCKETIZE(capital_gain, num_buckets=5), BUCKETIZE(capital_loss, num_buckets=5), BUCKTIZE(hours_per_week, num_buckets=6)) AS group_1, 8),
-    EMBEDDING(CONCAT(HASH(education), HASH(occupation), VOCABULARIZE(martial_status), VOCABULARIZE(relationship)) AS group_2, 8),
-    EMBEDDING(CONCAT(BUCKETIZE(age, num_buckets=5), HASH(native_country), VOCABULARIZE(race), VOCABULARIZE(sex)) AS group_3, 8)
-    FOR deep_embeddings
-COLUMN
+  pan(
+    EMBEDDING(
+        CONCAT(VOCABULARIZE(workclass), 
+               BUCKETIZE(capital_gain, num_buckets=5), 
+               BUCKETIZE(capital_loss, num_buckets=5), 
+               BUCKETIZE(hours_per_week, num_buckets=6)) AS group_1, 
+        8),
+    EMBEDDING(
+        CONCAT(HASH(education), 
+               HASH(occupation), 
+               VOCABULARIZE(martial_status), 
+               VOCABULARIZE(relationship)) AS group_2, 
+        8),
+    EMBEDDING(
+        CONCAT(BUCKETIZE(age, num_buckets=5), 
+               HASH(native_country), 
+               VOCABULARIZE(race), 
+               VOCABULARIZE(sex)) AS group_3, 
+        8)
+    ) AS deep_embeddings
+  pan(
     EMBEDDING(group1, 1),
     EMBEDDING(group2, 1)
-    FOR wide_embeddings
+  ) AS wide_embeddings
```

Comment:
After further consideration, users may be confusing about how many tensor the COLUMN expression outputs. Please check the following sample clauses, the styles are the same, but the expected outputs are different:

User expects the output is one tensor.

COLUMN age, STANDARDIZE(capital_gain), EMBEDDING(HASH(workclass))
User expected the output is two tensors.

COLUMN PAD(EMBEDDING(…), EMBEDDING(…)), PAD(EMBEDDING(…), EMBEDDING(…))

Reply:
After further discussion with more SQLFlow teammates, we choose this following syntax design: **keep one COLUMN clause**; If we want to output multiple tensors in this clause, use keyword `FOR` before the output tensor name. Please check the following example:

```
COLUMN
  pan(
    EMBEDDING(
        CONCAT(VOCABULARIZE(workclass), 
               BUCKETIZE(capital_gain, num_buckets=5), 
               BUCKETIZE(capital_loss, num_buckets=5), 
               BUCKETIZE(hours_per_week, num_buckets=6)) AS group_1, 
        8),
    EMBEDDING(
        CONCAT(HASH(education), 
               HASH(occupation), 
               VOCABULARIZE(martial_status), 
               VOCABULARIZE(relationship)) AS group_2, 
        8),
    EMBEDDING(
        CONCAT(BUCKETIZE(age, num_buckets=5), 
               HASH(native_country), 
               VOCABULARIZE(race), 
               VOCABULARIZE(sex)) AS group_3, 
        8)
    ) FOR deep_embeddings,
  pan(
    EMBEDDING(group1, 1),
    EMBEDDING(group2, 1)
  ) FOR wide_embeddings
LABEL label
```

In the COLUMN clause, we use two keyword `AS` and `FOR`. Why don't we merge them into one - `AS`? Because they have different meanings.
1. `AS` is the alias of a part of transform logic. The alias can be shared by multiple transform function call.
2. `FOR` means this part of transform logic will output a tensor. The IDENT after `FOR` is the tensor name.

Reply:
what's mean of `pan`?

Reply:
> what's mean of `pan`?

`pan` is a temporary name which means a transform function to merge multiple tensor into one tensor. We need figure out another more accurate name to describe the transform logic.